linux_docker_resources/Dockerfile-RHEL

```diff
@@ -175,6 +170,15 @@ RUN dnf install \
     yamllint \
     --refresh -y
 
+# Install dependencies for LTTng
+RUN if test \( ${ROS_DISTRO} = humble \); then \
+      true \
+    ; elif test \( ${ROS_DISTRO} = iron \); then \
+      dnf install lttng-ust-devel lttng-tools python3-babeltrace python3-lttng --refresh -y \
+    ; else \
+      dnf install lttng-ust-devel lttng-tools python3-babeltrace lttng-tools-devel --refresh -y \
+    ; fi
+
```

Comment:
So we don't need to check for `${EL_RELEASE/.*/} != 8` anymore?

Reply:
Right, good question.  Only `humble` uses RHEL-8, so the first part of the `if` stanza is a stand-in for what used to be our RHEL-8 checks.  All of Iron, Jazzy, and Rolling use RHEL-9, so they all have the rest of the packages available.

In some sense, I guess the logic here really is:

```
if ($OS == RHEL-8)
   # Do nothing, since the lttng packages are too old *and* humble doesn't support this out-of-the-box
elif ($OS == RHEL-9)
  if ($ROS_DISTRO == iron)
    dnf install lttng-ust-devel lttng-tools python3-babeltrace python3-lttng --refresh -y
  else
    dnf install lttng-ust-devel lttng-tools python3-babeltrace lttng-tools-devel --refresh -y
  fi
fi
```

Reply:
Right now I can trigger a RHEL 8 build of Rolling. This makes that impossible.

I don't really follow why Jazzy warrants these changes. Can't we just change the conditional on line 108 to include Jazzy as well and call it a day?

Reply:
> I'm lukewarm on this. If I understand it correctly, it couples the package logic between the ROS distro and the platform.

Yes and no.  It's complicated.

If we ignore the exact platform for the moment, the situation is this:
1.  Humble doesn't need any lttng packages in CI, because we don't support ros2_tracing out-of-the-box.
2.  Iron needs lttng-ust-devel, lttng-tools, python3-babeltrace, and python3-lttng, because there we use the Python bindings as provided by the platform.
3.  Rolling, Jazzy, and presumably newer needs lttng-ust-devel, lttng-tools, python3-babeltrace, and lttng-tools-devel because we build our own Python bindings.

If we then add in the platform to the mix, the situation is that:
1.  RHEL-8 can't possibly support any of this, because the version of lttng is too old (2.8) and our code doesn't support this old version.
2.  RHEL-9 can support any of the combinations of things that we need for any of the ROS distributions.

I'm not sure where that leaves us in terms of this logic.  Any suggestions on what we might do differently here?

Reply:
> I don't really follow why Jazzy warrants these changes. Can't we just change the conditional on line 108 to include Jazzy as well and call it a day?

We can, but then we'll run into the same problem when we get to K-Turtle.  I was trying to make it so we didn't have to think about this in the future.

> Right now I can trigger a RHEL 8 build of Rolling. This makes that impossible.

True, but I highly doubt that would work for other reasons anyway (python version, etc).

Reply:
>  I was trying to make it so we didn't have to think about this in the future.

Alright, then let's drop the `-a ${UBUNTU_DISTRO} = jammy` entirely. If Iron don't build when `lttng-tools-devel` is installed, we're boxing ourselves out of building multiple ROS distributions on the same host anyway.

Reply:
> Alright, then let's drop the `-a ${UBUNTU_DISTRO} = jammy` entirely. If Iron don't build when `lttng-tools-devel` is installed, we're boxing ourselves out of building multiple ROS distributions on the same host anyway.

All right, fair enough.  I'll change this up (I'll have to do the same thing with python3-lttng, but that should be fine).

Reply:
Actually, let's start a database to hold all of this inter-platform dependency logic. We could even whip up a tool to automatically invoke the system's package manager to install the packages automatically.

/s ...kinda.

Reply:
> Actually, let's start a database to hold all of this inter-platform dependency logic. We could even whip up a tool to automatically invoke the system's package manager to install the packages automatically.

Heh, yeah.  I've actually tried to refactor this code to support `rosdep`, but it would be a big change; we'd have to move all of the logic for dealing with `ros2.repos` into the Dockerfiles so we could then run `rosdep` in there.  And then we'd have to figure out how to do the equivalent in Windows.  It isn't impossible, but it also isn't trivial.

Reply:
FWIW, that's how the CI jobs on build.ros2.org work, and they still have docker caching.